frontend/src/app/store/stored.ts

```diff
@@ -0,0 +1,103 @@
+import { Params } from '@angular/router';
+import { Store } from './types';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { distinctUntilChanged, map, takeUntil } from 'rxjs/operators';
+import * as _ from 'lodash';
+
+
+
+export abstract class Stored<State extends object> {
+    /**
+     * The stored state to of the model.
+     *
+     * This is implemented as a BehaviorSubject so you can read state$.value, but
+     * you should NOT push values onto the state directly. Use setParams() instead.
+     */
+    state$: BehaviorSubject<State>;
+
+    protected complete$ = new Subject<void>();
```

Comment:
Upon first read of documentation and classes, I find it hardest to grasp what `Stored` is for, and what it does. Perhaps it would be clearer if you called this `StoreDirective`? Because this is what components will have to extend, right? I think adopting Angular-speak will help future developers.

Reply:
Could we reimplement this as a directive, rather than a model?